torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -158,8 +169,13 @@ Tensor norm_backward(const Tensor & grad, const Tensor & self, const optional<Sc
     self_scaled = self;
     scale_v = grad / norm;
   } else if (std::isinf(p)) {
-    self_scaled = self.sign() * (self.abs() == norm).type_as(self);
-    scale_v = grad.clone(at::MemoryFormat::Preserve);
+    Tensor is_eq_max = (self.abs() == norm).logical_or_(self.isnan().logical_and_(norm.isnan())).type_as(self);
+    self_scaled = self.sign() * is_eq_max;
+    Tensor nb_max = is_eq_max.count_nonzero(dim);
+    if (self.dim() != 0) {
```

Comment:
nit: given the condition for the other unsqueeze, shouldn't this one have a `!keepdim` as well?

Reply:
The condition for the grad and norm unsqueeze is there because grad and norm are pre-unsqueezed if keepdim is True. `nb_max` on the other hand is the output of count_nonzero, which always collapse the dims. `nb_max` therefore always requires an unsqueeze in order to match the number of dims of grad.

Reply:
Sounds good!